nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/trash/test/TestTrashBackwardMechanism.java

```diff
@@ -62,19 +63,55 @@ public void testPropertyTrashService() {
 
     @Test
     public void testMechanism() {
-        DocumentModel doc = session.createDocumentModel("/", "file001", "File");
-        doc = session.createDocument(doc);
+        DocumentModel folder = session.createDocumentModel("/", "folder", "Folder");
+        DocumentModel file1 = session.createDocumentModel("/folder", "file1", "File");
+        DocumentModel file2 = session.createDocumentModel("/folder", "file2", "File");
+        folder = session.createDocument(folder);
+        file1 = session.createDocument(file1);
+        file2 = session.createDocument(file2);
         session.save();
 
         // follow delete transition to check if TrashService is called
-        session.followTransition(doc, LifeCycleConstants.DELETE_TRANSITION);
```

Comment:
Why removing this? This whole test class is about testing the backward mechanism on `deleted` lifecycle state and it's not intended to test anything else. I think your test would be better in the property trash service test class.

Reply:
Indeed wrong test class. Now the code I added is in the `AbstractTrashService`, maybe it's better to put the test in the `AbstractTestTrashService` ?

Reply:
Ah yes indeed 👍